### PR TITLE
[Feature] Added routes to get repos from gitlab

### PR DESCRIPTION
**Pull request by r-scheele**
- Requires user login to gitlab, to generate access token for private repos
- Example response;
```json
[
  {
    "name": "opal-policy-example",
    "html_url": "https://gitlab.com/r-scheele/opal-policy-example",
    "owner": "r-scheele"
  }
]
```
- Updated image on dockerhub

### CODE REVIEW

File: app/server/auth/authenticate.py [hunk 1]
```diff
@@ -12,6 +12,16 @@
...
+@router.get("/gitlab/login")
+async def login_gitlab():
+    APP_ID, REDIRECT_URI = (
+        settings.GITLAB_CLIENT_ID,
+        "http://localhost:8080/api/auth/callback/gitlab",
+    )
+    url = f"https://gitlab.com/oauth/authorize?client_id={APP_ID}&redirect_uri={REDIRECT_URI}&response_type=code"
+    return RedirectResponse(url=url)
+
+
```
Comment (Youngestdev) [inline app/server/auth/authenticate.py:24]: We can remove this now. I have figured a way from the FE.

Comment (Youngestdev) [inline app/server/auth/authenticate.py:24]: We can remove this now. I have figured a way from the FE.

File: app/server/auth/authenticate.py [hunk 2]
```diff
@@ -12,6 +12,16 @@
...
 @router.get("/api/auth/callback/gitlab")
```
Comment (Youngestdev) [inline app/server/auth/authenticate.py:25]: Can we rename this to `/gitlab/token` instead? There's no uniformity in the API routes.

So we can have `/github/token` and `/gitlab/token`? Then split both routes so on the Swagger page they can be differentiated. Example:

# Token

/gitlab/token
/github/token

# Policy Operation

/policies*

# Data operation

/ data

# Repo management


/gitlab/repo
/github/repo

---

As we progress later on, we can update the functions so we just have a single route with an extra parameter `git_provider` that enables the function/method return the right output.

File: app/server/user.py [hunk 3]
```diff
@@ -25,3 +25,24 @@ async def get_public_and_private_repo(
...
+    ).json()
+
+    return [
+        {
+            "name": repo["name"],
+            "html_url": repo["web_url"],
+            "owner": repo["owner"]["username"],
+        }
+        for repo in repos
+    ]
```
Comment (Youngestdev) [inline app/server/user.py:48]: Can we do this expression outside the return? Maybe:

```py
repos = [ x for x in repos ]

return repos
```

**Review by Youngestdev — COMMENTED** [review]
Awesome work. Let's adjust the changes I mentioned and we should be good to go.

File: app/server/auth/authenticate.py (hunk 1 above)
Comment (r-scheele) [reply app/server/auth/authenticate.py:24]: That's great boss

**Review by Youngestdev — APPROVED** [review]

**Pull request merged**